lib/dnsimple/error.rb

```diff
@@ -9,8 +9,19 @@ class RequestError < Error
 
     def initialize(http_response)
       @http_response = http_response
-      super(http_response.code.to_s)
+      super(message_from(http_response))
     end
+
+    private
+
+    def message_from(http_response)
+      if http_response.headers["Content-Type"] == "application/json"
+        http_response.parsed_response["message"]
+      else
+        http_response.code
```

Comment:
@jacegu is there a way to generate a message like

    HTTP 502: Bad Gateway

or at least

    HTTP 502

Just the code looks quite hard to immediately understand.